libs/blocks/modal/modal.js

```diff
@@ -137,7 +152,12 @@ export async function getModal(details, custom) {
     [...document.querySelectorAll('header, main, footer')]
       .forEach((element) => element.setAttribute('aria-disabled', 'true'));
   }
-
+  if (dialog.classList.contains('commerce-frame')) {
+    const { debounce } = await import('../../utils/action.js');
+    window.addEventListener('message', (messageInfo) => {
+      debounce(sendViewportDimensionsOnRequest(messageInfo));
+    });
```

Comment:
You only want to set this up once, otherwise u will end up creating multiple resize event listeners

Reply:
The code still doesn't look correct. Everytime you get a `message` event you will call the `sendViewportDimensionsOnRequest` method, which will set up a new resize event listener without discarding the old one.

If you receive 10 messages, you have 10 resize event listeners and call `sendViewportDimensionsToiFrame` 10 times.

Reply:
You have a couple of options to fix that. Here are two suggestions:
1. Remove the 'message' event listener after the initial set up OR
2. Check if there is already a resize event listener, if yes - don't set it up again